drivers/npmx/npmx_shell.c

```diff
@@ -1462,795 +898,984 @@ static int buck_gpio_set(const struct shell *shell, size_t argc, char **argv,
-		shell_error(shell, "Error: invalid LDSW mode (0 -> LOADSW, 1 -> LDO).");
+	npmx_error_t err_code = npmx_charger_trickle_voltage_set(charger_instance, charger_trickle);
+	if (!check_error_code(shell, err_code)) {
+		print_set_error(shell, "trickle voltage");
 		return 0;
 	}
 
-	npmx_ldsw_t *ldsw_instance = npmx_ldsw_get(npmx_instance, ldsw_indx);
-	npmx_error_t err_code = npmx_ldsw_mode_set(ldsw_instance, mode);
+	print_usuccess_unit(shell, voltage_mv, "mV");
+	return 0;
+}
 
-	if (err_code != NPMX_SUCCESS) {
-		shell_error(shell, "Error: unable to set LDSW mode.");
-	}
+static int cmd_charger_trickle_voltage_get(const struct shell *shell, size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
 
-	/* LDSW reset is required to apply mode change. */
-	err_code = npmx_ldsw_task_trigger(ldsw_instance, NPMX_LDSW_TASK_DISABLE);
+	npmx_charger_t *charger_instance = charger_instance_get(shell, NULL, false);
+	if (charger_instance == NULL) {
+		return 0;
+	}
 
+	npmx_charger_trickle_t charger_trickle;
+	npmx_error_t err_code =
+		npmx_charger_trickle_voltage_get(charger_instance, &charger_trickle);
 	if (!check_error_code(shell, err_code)) {
-		shell_error(shell,
-			    "Error: reset error while disabling specified LDSW to change mode.");
+		print_get_error(shell, "trickle voltage");
+		return 0;
 	}
 
-	err_code = npmx_ldsw_task_trigger(ldsw_instance, NPMX_LDSW_TASK_ENABLE);
-
-	if (check_error_code(shell, err_code)) {
-		shell_print(shell, "Success: %d.", (uint8_t)mode);
-	} else {
-		shell_error(shell,
-			    "Error: reset error while enabling specified LDSW to change mode.");
+	uint32_t voltage_mv;
+	if (!npmx_charger_trickle_convert_to_mv(charger_trickle, &voltage_mv)) {
+		print_convert_error(shell, "trickle voltage", "millivolts");
+		return 0;
 	}
 
+	print_uvalue_unit(shell, voltage_mv, "mV");
 	return 0;
 }
 
-static int cmd_ldsw_mode_get(const struct shell *shell, size_t argc, char **argv)
+static int charger_die_temp_set(const struct shell *shell, size_t argc, char **argv,
+				npmx_error_t (*func)(npmx_charger_t const *p_instance,
+						     int16_t temperature))
 {
-	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+	args_info_t args_info = { .expected_args = 1,
+				  .arg = {
+					  [0] = { SHELL_ARG_TYPE_INT_VALUE,
+						  "die temperature threshold" },
+				  } };
+	if (!arguments_check(shell, argc, argv, &args_info)) {
+		return 0;
+	}
 
-	if (npmx_instance == NULL) {
-		shell_error(shell, "Error: shell is not initialized.");
+	npmx_charger_t *charger_instance =
+		charger_instance_get(shell, "die temperature threshold", true);
+	if (charger_instance == NULL) {
 		return 0;
 	}
 
-	if (argc < 2) {
-		shell_error(shell, "Error: missing LDSW instance index.");
+	int16_t temperature = (uint16_t)CLAMP(args_info.arg[0].result.ivalue,
+					      NPMX_PERIPH_CHARGER_DIE_TEMPERATURE_MIN_VAL,
+					      NPMX_PERIPH_CHARGER_DIE_TEMPERATURE_MAX_VAL);
```

Comment:
CLAMP should not be used to align values to the allowed range. I added a description to [Writing npmx shell](https://nordicsemi.atlassian.net/wiki/spaces/NRFX/pages/301893654/Writing+npmx+shell)
Please, check if other uses of CLAMP are as they are supposed to be.